dali/operators/util/randomizer.cuh

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef DALI_OPERATORS_UTIL_RANDOMIZER_CUH_
+#define DALI_OPERATORS_UTIL_RANDOMIZER_CUH_
+
+#include <math.h>
+#include "dali/core/device_guard.h"
+#include "dali/pipeline/data/backend.h"
+#include <curand_kernel.h>  // NOLINT
+
+namespace dali {
+
+namespace detail {
+
+template <typename T>
+struct CurandNormal {};
+
+template <>
+struct CurandNormal<float> {
+  __device__ static float normal(curandState *state) {
+    return curand_normal(state);
+  }
+};
+
+template <>
+struct CurandNormal<double> {
+  __device__ static double normal(curandState *state) {
```

Comment:
DALI_HOST_DEV DALI_FORCEINLINE ??

Reply:
This is a `__device___` only function, so I am not sure if `DALI_HOST_DEV` should be used here.

Reply:
Added `DALI_FORCEINLINE`